cli/src/pcluster/config/cluster_config.py

```diff
@@ -1449,3 +1490,23 @@ def _register_validators(self):
                     # FIXME: head_node.architecture vs compute_resource.architecture?
                     architecture=self.head_node.architecture,
                 )
+
+    @property
+    def image_dict(self):
+        """Return image dict of queues, key is queue name, value is image id."""
+        if self.__image_dict:
+            return self.__image_dict
+        self.__image_dict = {}
+        ami_filters = self.dev_settings.ami_search_filters if self.dev_settings else None
+
+        for queue in self.scheduling.queues:
+            if queue.queue_ami_id:
+                self.__image_dict[queue.name] = queue.queue_ami_id
+            elif self.image.custom_ami:
+                self.__image_dict[queue.name] = self.image.custom_ami
+            else:
+                self.__image_dict[queue.name] = AWSApi.instance().ec2.get_official_image_id(
```

Comment:
let's get the value of the property `headnode_ami_id` otherwise we are going to call `get_official_image_id` N times to return always the same value